scripts/build/termux_get_repo_files.sh

```diff
@@ -1,64 +1,85 @@
 termux_get_repo_files() {
+	local PACKAGES_HASH RELEASE_FILE repo_base dl_prefix RELEASE_FILE_URL RELEASE_FILE_SIG_URL
+	local -a pids=()
 	# Not needed for on-device builds or when building
 	# dependencies.
-	if [ "$TERMUX_ON_DEVICE_BUILD" = "true" ] || [ "$TERMUX_INSTALL_DEPS" = "false" ]; then
+	if [[ "$TERMUX_ON_DEVICE_BUILD" = "true" || "$TERMUX_INSTALL_DEPS" = "false" ]]; then
 		return
 	fi
 
-	for idx in $(seq ${#TERMUX_REPO_URL[@]}); do
-		local TERMUX_REPO_NAME=$(echo ${TERMUX_REPO_URL[$idx-1]} | sed -e 's%https://%%g' -e 's%http://%%g' -e 's%/%-%g')
-		if [ "$TERMUX_REPO_PKG_FORMAT" = "debian" ]; then
-			local RELEASE_FILE=${TERMUX_COMMON_CACHEDIR}/${TERMUX_REPO_NAME}-${TERMUX_REPO_DISTRIBUTION[$idx-1]}-Release
-			local repo_base="${TERMUX_REPO_URL[$idx-1]}/dists/${TERMUX_REPO_DISTRIBUTION[$idx-1]}"
-			local dl_prefix="${TERMUX_REPO_NAME}-${TERMUX_REPO_DISTRIBUTION[$idx-1]}-${TERMUX_REPO_COMPONENT[$idx-1]}"
-		elif [ "$TERMUX_REPO_PKG_FORMAT" = "pacman" ]; then
-			local JSON_FILE="${TERMUX_COMMON_CACHEDIR}-${TERMUX_ARCH}/${TERMUX_REPO_NAME}-json"
-			local repo_base="${TERMUX_REPO_URL[$idx-1]}/${TERMUX_ARCH}"
-		fi
+	case "$TERMUX_REPO_PKG_FORMAT" in
+		"debian" | "pacman");;
+		*) termux_error_exit "Invalid package format: $TERMUX_REPO_PKG_FORMAT"
+	esac
+
+	for idx in "${!TERMUX_REPO_URL[@]}"; do
+		local TERMUX_REPO_NAME="${TERMUX_REPO_URL[$idx]#https://}"
+		TERMUX_REPO_NAME="${TERMUX_REPO_NAME#http://}"
+		TERMUX_REPO_NAME="${TERMUX_REPO_NAME//\//-}"
+		case "$TERMUX_REPO_PKG_FORMAT" in
+		"debian")
+			RELEASE_FILE="${TERMUX_COMMON_CACHEDIR}/${TERMUX_REPO_NAME}-${TERMUX_REPO_DISTRIBUTION[$idx]}-Release"
+			repo_base="${TERMUX_REPO_URL[$idx]}/dists/${TERMUX_REPO_DISTRIBUTION[$idx]}"
+			dl_prefix="${TERMUX_REPO_NAME}-${TERMUX_REPO_DISTRIBUTION[$idx]}-${TERMUX_REPO_COMPONENT[$idx]}"
+			RELEASE_FILE_URL="${repo_base}/Release"
+			RELEASE_FILE_SIG_URL="${RELEASE_FILE_URL}.gpg"
+			;;
+		"pacman")
+			RELEASE_FILE="${TERMUX_COMMON_CACHEDIR}-${TERMUX_ARCH}/${TERMUX_REPO_NAME}-json"
+			repo_base="${TERMUX_REPO_URL[$idx]}/${TERMUX_ARCH}"
+			RELEASE_FILE_URL="${repo_base}/${TERMUX_REPO_DISTRIBUTION[$idx]}.json"
+			RELEASE_FILE_SIG_URL="${RELEASE_FILE_URL}.sig"
+			;;
```

Comment:
I meant can't you put the exit here if we're already casing on `$TERMUX_REPO_FORMAT`.
```suggestion
			;;
		*) termux_error_exit "Invalid package format: $TERMUX_REPO_PKG_FORMAT";;
```
Also you missed the `;;` in the one above.

Reply:
Does that really matter that much?
If it's nonsensical it'll bail on the first iteration of the loop.

Reply:
Oh, it is not in background subshell. But still in a loop.
I think it is fine to add that.

Reply:
Done.